### DISCUSSION TITLE
What am I doing wrong?

### DISCUSSION THREAD

**Discussion by Denny09310** [IEvangelist/blazorators#53] [Q&A]
Hi, I came across the source generator library yesterday, seeing the potential of it I was tempted to try it. I needed to implement the Bluetooth API, so I tried the tutorial but soon enough saw that was impossible because in lib.dom.d.ts the Bluetooth interface wasn't present and obviously it can't be scaffolded. At this point I discovered the branch with the "new-parser". Navigating through the code, even to offer some help when (and if 😁) I would understood the code, I found the "TypeDeclarationSources" property and the commented code to gather that raw file with the http client. I made some simple modifications to try make it work, but when I try to start the BlazorExampleConsumer project the compilation get stuck, using the detailed verbosity I saw that it gets stuck at the "CSC" activity. So to sum up the problem is that after forked, cloned and installed all the necessary tools the project won't compile. Is the compilation taking really much time? Or there is some problem behind? Because I don't have a bad PC so it's strange to me, but it's like 10 minutes compiling now. Thanks in advance to anyone who can help me. If you need further logs ask me without problem

**Comment by IEvangelist** [comment]
Hey there! Thank you for your interest in my project... Yeah, this is certainly limited. I have been working on a separate branch for a while now, with little progress lately (due to time constraints). The idea for the branch is to introduce a means of pointing to custom TypeScript declarations. As of now, it's still not officially supported. Furthermore, even for the types that are in the `lib.dom.d.ts` we only cover a certain subset of them. Again, it's something I'd love to add support for, but I realistically haven't had the time.

**Reply by Denny09310** [reply]
No worries, as a matter of fact I want to help where I can. The problem I'm experiencing is that even with 15 minutes of compilation time it doesn't start the sample project and I simply want to know if my PC sucks or there's something I'm doing wrong 😂. Even deleting and re-cloning the project don't work. I've noticed that it max out all the cores of my CPU, so I think something is doing, but realistically 10/15 minutes is weird. I can attach my task manager screenshot as proof and system specs.

![2024-06-03](https://github.com/IEvangelist/blazorators/assets/50493437/e43b7adf-39d0-4b78-8dc1-026ee42ea19c)

**Reply by IEvangelist** [reply]
Wow, I must have overlooked that part. I haven't experienced those long build issues, I wonder if there's not something else going on there.... That's crazy, so what exactly did you do to get to that state? Also, here's the branch I have that I was working on, just have a look at all the updates: [see the `ast` branch](https://github.com/IEvangelist/blazorators/compare/main...ast).

**Reply by Denny09310** [reply]
the ast branch, has https://github.com/IEvangelist/blazorators/commit/6c3ad1e63466ff0febb76ba9c4eae3c0548d059b that introduced lot's of errors, I've tried fix them but it's taking too long and I don't know if it compiles either. I'll try reinstall visual studio and dotnet alongside the necessary tools and try again building at least the main branch. Thanks for the help 😁

**Reply by IEvangelist** [reply]
Sorry, that was the wrong branch... the branch I meant to say was a dead one, I deleted it. The branch is `new-parser`... https://github.com/IEvangelist/blazorators/compare/main...new-parser This one had unit tests that work with the new parser, I just needed to start mapping out the functionality of taking the AST and converting it into services, etc.

**Reply by Denny09310** [reply]
Ok, so just as info. I tried compiling the projects one by one and I've noticed that the only one not compiling was in fact the sample. So after compiling the test I ran them and it's working 🎉. Not sure what is the problem with the sample but with the tests there are no issues. The only thing I've noticed even in the tests is that it's blocking in the "ReadTypeDeclarationMap" function while performing the Parallel ForEach. Maybe it wasn't compiling the error, it was this analysis
